trio/_core/_asyncgens.py

```diff
@@ -22,17 +39,21 @@ class AsyncGenerators:
     # asyncgens after the system nursery has been closed, it's a
     # regular set so we don't have to deal with GC firing at
     # unexpected times.
-    alive = attr.ib(factory=weakref.WeakSet)
+    alive: weakref.WeakSet[AsyncGeneratorType[object, NoReturn]] | set[
```

Comment:
Since you made the type aliases above, perhaps use those here - would let this fit on one line.

Reply:
They are not aliases, they are purely for being able to set the factory type so there are not type errors

Reply:
Well they function as type aliases, so you could use them as such.

Reply:
+1

Reply:
Changed in d2c5f3c